Zend/zend_API.c

```diff
@@ -539,7 +555,7 @@ static const char *zend_parse_arg_impl(int arg_num, zval *arg, va_list *va, cons
 					break;
 				} else {
 					if (is_callable_error) {
-						*severity = E_WARNING;
+						*severity = strict ? (E_EXCEPTION | E_ERROR) : E_WARNING;
```

Comment:
This branch should be redundant. Since `error` is always populated, `zend_parse_arg` will always go into the the error mode and ignore the `severity` argument. The only time it won't is if `quiet` is set. At which point it will check if `severity` is `E_STRICT` to determine failure.

So this could be replaced with `E_ERROR` for the same functionality, reducing the number of branches.